homeassistant/components/alarmdecoder/binary_sensor.py

```diff
@@ -112,6 +116,13 @@ async def async_added_to_hass(self):
             )
         )
 
+        def set_default_state(_):
+            if self._attr_is_on is None:
+                self._attr_is_on = False
+                self.schedule_update_ha_state()
+
+        async_call_later(self.hass, DEFAULT_STATE_TIMEOUT, set_default_state)
```

Comment:
Why even call it on unknown initially?

Reply:
I'd just set it to False from the beginning.

Reply:
The idea was to avoid possible false negative state updates.

Reply:
Why then update after 30 seconds? That's also a false negative.

Reply:
As I understand the device/API, it will send explicit messages when a zone (represented by a `binary_sensor` on our side) is faulted (we set to `True`). However if a zone is _not_ faulted we will receive no messages to affirm this. If a zone is already faulted before HA startup or a faulted zone is configured during runtime, then we can be fairly confident that 30s is enough time to receive the initial fault message.

Reply:
Will the api send a faulted state when we connect, and there's an existing fault, or just when the fault happens?

Reply:
Based on my testing it appears to cycle through the faulted zones repeatedly while they're faulted, but I would love to have someone more familiar with the API chime in here.

Reply:
This works well in my local testing but perhaps it should get a maintainer review first.

Reply:
If there's an ongoing reporting of faulted states, this PR is good. It would be good to confirm the cadence of those reports though so we can set the time accordingly.

Reply:
I have one of these (and have coded for it some) and I don't believe it reports continuously. The actual alerts and sensor state changes are notified once when they occur.  There is code in the `alarmdecoder` package that only reports on state changes.    It sends the panel message (what's displayed on the keypads) repeatedly but that's just a string sensor.  So you'll see messages saying certain zones are faulted or the alarm is sounding but the actual state triggers only occur once.

IMO, the best way to handle this is to change the component to inherit from `RestoreEntity` so the last states would be retained.  Yes - it's possible you'll miss states and be wrong when HA is off - but the current sensor states are wrong 100% of the time at startup.  Without the ability to poll the alarm panel (which the hardware doesn't support), I think this is the best you can do.  If you decide not to do that, I'd say just set the initial state to False.  Waiting 30 sec doesn't add any functionality that I can see.  If you get an update in the first 30 sec that's going to be used in either case so why make it more complicated.

Reply:
I just reviewed the alarmdecoder documentation and I was wrong earlier, the state of the zone/sensor is not reported when the "attributes" message arrives. The occasional update that the RF devices provide only contains loop and battery info. The state is only updated when the status of the binary sensor is toggled. I also believe the RestoreEntity is the way to go. It will miss any changes while homeassistant is down and will still update when the zone changes, better than all of the entities being unknown after every reboot. 
Aside from that, since all of them are currently in this state and I would assume that they would just restore to that state if we use that method, it might be a good idea to set them all to false once so that binary sensors which can't be toggled get a state (think gas sensors and smoke alarms).

Reply:
I just created a node-red flow to handle it like this: if binary sensor state = unknown AND alarm panel display = disarmed then set state to off. Given it's winter now and most of my zones are always off this will work for now. 
I'm not sure if there is a way to capture a list of "faults" from the alarm display entity and then set the binary sensors accordingly ie if the zone is in the list of faults set the binary sensor to on, else set to off. 
That might be the best route to make sure that all the states are accurate, even if homeassistant had been offline for a while. But this would be way beyond my python skills.